### PR TITLE
Encode uint64 as string on the wire via moshi

### PR DESCRIPTION
**Pull request by oldergod**
part of #1387

Just doing what protoc is.

We have a change in the way we print json but the parsing isn't affected so maybe that's fine?

- [ ] support int64, fixed64, uint64
- [ ] tests that confirm we handle unquoted strings (and confirming it's consistent with protoc)
- [ ] check if protos support weird stuff like `{ orders: "123.0" } or { orders: "123e1" }`

### CODE REVIEW

File: wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt [hunk 1]
```diff
@@ -135,10 +135,10 @@ class WireJsonAdapterFactory private constructor(
       @Throws(IOException::class)
       override fun toJson(writer: JsonWriter, value: Long?) {
         if (value!! < 0) {
-          val unsigned = power64.add(BigInteger.valueOf(value))
+          val unsigned = power64.add(BigInteger.valueOf(value)).toString()
```
Comment (swankjesse) [inline wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt:138]: I don’t think this is correct

Comment (swankjesse) [reply wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt:138]: Though perhaps in this incorrectness it is consistent with proto3? What do the docs advise?

**Comment by swankjesse** [comment]
Hold the phone. We need to use strings for these three: int64, fixed64, uint64:
https://developers.google.com/protocol-buffers/docs/proto3#json

**Comment by oldergod** [comment]
will come back. I think I need to fix wireField first.

**Pull request closed without merging**